collision.go

```diff
@@ -16,6 +16,16 @@ type SpaceComponent struct {
 	Height   float32
 }
 
+// Center positions the space component according to its center instead of its
+// top-left point (this avoids doing the same math each time in your systems)
+func (sc *SpaceComponent) Center(p Point) {
```

Comment:
Hmm. I'm not sure this is the best way to approach this problem.

For example:

**Sprite A**:

- Position: `0, 0`
- System: `DoesNotCallCenter`
- Width: `20`


**Sprite B**:

- Position: `0, 0`
- System: `CallsCenter`
- Width: `20`

If this code was run, **Sprite B** would end up at position `(10, 10)`, while **Sprite A** kept the same position - Even if the system made no other changes.